stripe/src/main/java/com/stripe/android/paymentsheet/BottomSheetController.kt

```diff
@@ -5,60 +5,51 @@ import android.view.ViewGroup
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.distinctUntilChanged
 import com.google.android.material.bottomsheet.BottomSheetBehavior
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.delay
-import kotlinx.coroutines.launch
 
 internal class BottomSheetController(
-    private val bottomSheetBehavior: BottomSheetBehavior<ViewGroup>,
-    private val lifecycleScope: CoroutineScope
+    private val bottomSheetBehavior: BottomSheetBehavior<ViewGroup>
 ) {
     private val _shouldFinish = MutableLiveData(false)
     internal val shouldFinish = _shouldFinish.distinctUntilChanged()
 
     fun setup() {
         bottomSheetBehavior.isHideable = true
         bottomSheetBehavior.isDraggable = false
-        // Start hidden and then animate in after delay
         bottomSheetBehavior.state = BottomSheetBehavior.STATE_HIDDEN
 
-        lifecycleScope.launch {
-            delay(ANIMATE_IN_DELAY)
-
-            bottomSheetBehavior.state = BottomSheetBehavior.STATE_EXPANDED
-            bottomSheetBehavior.addBottomSheetCallback(
-                object : BottomSheetBehavior.BottomSheetCallback() {
-                    override fun onSlide(bottomSheet: View, slideOffset: Float) {
-                    }
+        bottomSheetBehavior.addBottomSheetCallback(
+            object : BottomSheetBehavior.BottomSheetCallback() {
+                override fun onSlide(bottomSheet: View, slideOffset: Float) {
+                }
 
-                    override fun onStateChanged(bottomSheet: View, newState: Int) {
-                        when (newState) {
-                            BottomSheetBehavior.STATE_EXPANDED -> {
-                                // Because we change the content of the sheet and its height at
-                                // runtime, make sure it's properly laid out once it settles
-                                bottomSheet.requestLayout()
-                            }
-                            BottomSheetBehavior.STATE_HIDDEN -> {
-                                // finish the activity only after the bottom sheet's state has
-                                // transitioned to `BottomSheetBehavior.STATE_HIDDEN`
-                                _shouldFinish.value = true
-                            }
-                            else -> {
-                            }
+                override fun onStateChanged(bottomSheet: View, newState: Int) {
+                    when (newState) {
+                        BottomSheetBehavior.STATE_EXPANDED -> {
+                            // isFitToContents causes conflicts when calculating the sheet position
+                            // upon resize. CoordinatorLayout will already position the sheet
+                            // correctly with gravity = bottom.
+                            bottomSheetBehavior.isFitToContents = false
+                        }
```

Comment:
It needs to be deferred until expanded. It must be true when transitioning from STATE_HIDDEN to STATE_EXPANDED, otherwise it takes the full screen as it goes through STATE_SETTLING.